### PR TITLE
Handle memory view serialization in data frame

### PR DESCRIPTION
**Pull request by fredthomsen**
Help address issue 250 in hyper-h2

### CODE REVIEW

**Comment by fredthomsen** [comment]
@Lukasa added.  Thanks.

File: HISTORY.rst [hunk 1]
```diff
@@ -4,6 +4,10 @@ Release History
 6.0.0dev0
 ---------
 
+**API Changes**
```
Comment (Lukasa) [inline HISTORY.rst]: This should be `**API Changes (Backward-compatible)**` as you can see a few paragraphs below. =)

Comment (fredthomsen) [inline HISTORY.rst]: Silly me. Added.

File: test/test_frames.py [hunk 2]
```diff
@@ -113,10 +113,14 @@ def test_data_frame_has_correct_flags(self):
             'END_STREAM', 'PADDED'
         ])
 
-    def test_data_frame_serializes_properly(self):
+    @pytest.mark.parametrize('data', [
+        b'testdata',
+        memoryview(b'testdata')
+        ])
```
Comment (alexwlchan) [inline test/test_frames.py]: I don’t know how much @Lukasa cares, but one tiny nitpick is that these brackets should be dedented once to match the bracing style in the rest of the project.

Comment (fredthomsen) [inline test/test_frames.py]: Have to keep it clean. Done.  I need to adjust my flake8 settings. I have too many warnings on right now and that makes me ignore some of them.

**Pull request merged**